packages/fx-core/src/plugins/resource/frontend/index.ts

```diff
@@ -44,91 +45,70 @@ export class FrontendPlugin implements Plugin {
   frontendPluginImpl = new FrontendPluginImpl();
   dotnetPluginImpl = new DotnetPluginImpl();
 
+  private getImpl(ctx: PluginContext): PluginImpl {
+    return FrontendPlugin.isVsPlatform(ctx) ? this.dotnetPluginImpl : this.frontendPluginImpl;
```

Comment:
Nit: Another direction may be considered is to use `Polymorphisn` if it's feasible (frontendPluginImpl shares the same interface with dotnetPluginImpl).

Reply:
I prefer to make FrontendPlugin stateless. Using polymorphisn make a state, that means pluginImpl either to be FrontendPluginImpl or DotnetPluginImpl